tests/test_scripts.py

```diff
@@ -54,6 +55,21 @@ def test_load_into_counting():
     assert os.path.exists(outfile)
 
 
+def test_load_into_counting_nonwritable():
+    script = scriptpath('load-into-counting.py')
+    args = ['-x', '1e3', '-N', '2', '-k', '20', '-t']
+
+    outfile = utils.get_test_data('test-nonwritable')
+    os.chmod(outfile, stat.S_IWOTH | stat.S_IRUSR)
```

Comment:
The only way I can find to do that is the git-cache-metadata thing. It
appears git itself doesn't normally track file permissions.

On Sat, Apr 4, 2015, 16:17 Michael R. Crusoe <notifications@github.com>
wrote:

> In tests/test_scripts.py
> <https://github.com/ged-lab/khmer/pull/892#discussion_r27771761>:
>
> > @@ -54,6 +55,21 @@ def test_load_into_counting():
> >      assert os.path.exists(outfile)
> >
> >
> > +def test_load_into_counting_nonwritable():
> > +    script = scriptpath('load-into-counting.py')
> > +    args = ['-x', '1e3', '-N', '2', '-k', '20', '-t']
> > +
> > +    outfile = utils.get_test_data('test-nonwritable')
> > +    os.chmod(outfile, stat.S_IWOTH | stat.S_IRUSR)
>
> Sure. Either create a new file and set its bits locally or store a file in
> git with the desired permissions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ged-lab/khmer/pull/892/files#r27771761>.
>